### ISSUE TITLE
Clean up poor organization in FilePath/ZipPath test suite

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6788]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#6788|
|Type|enhancement|
|Created|2013-10-14 13:28:09Z|
|Branch|https://github.com/twisted/twisted/tree/filepath-tests-6788|

These test suites have some mess that it would be nice to clean up:

  1. `test_paths` and `test_zippath` duplicate the `verifyObject(IFilePath, ...)` test.
  1. A number of `setContent` tests exercising different scenarios are mixed together on a `TestCase` with tests for other methods (making it difficult to see the complete set of `setContent` tests).
  1. Several test methods really contain multiple tests

<details><summary>Searchable metadata</summary>

```
trac-id__6788 6788
type__enhancement enhancement
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__branches_filepath_tests_6788 branches/filepath-tests-6788
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1381757289000000 1381757289000000
changetime__1382276693000000 1382276693000000
version__None None
owner__exarkun exarkun

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [40322]) Branching to 'filepath-tests-6788'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun removed owner|
|-|-|

The branch addresses these issues.

**Comment by twisted-trac** [comment]
|[<img alt="wallrj's avatar" src="https://avatars.githubusercontent.com/u/978965?s=50" width="50" height="50">](https://github.com/wallrj)|@wallrj set owner to @wallrj<br>@wallrj set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="wallrj's avatar" src="https://avatars.githubusercontent.com/u/978965?s=50" width="50" height="50">](https://github.com/wallrj)<a name="note_4"></a>|@wallrj set owner to @exarkun<br>@wallrj set status to `new`|
|-|-|

Thanks exarkun,

The test reorganisation looks nice. I expect it will be useful for [#4360](https://github.com/twisted/twisted/issues/4360). Is
that what you had in mind?

Points:
 1. typo "extensikon"
 2. Should there also be a test to show that the temporary file is left behind
    if there is an IOError during writing or renaming? The docstring for
    "test_customExtension" describes that, but doesn't actually test it. But it
    was never tested so we can save that for another ticket I suppose.
 3. Also there seems to be a windows specific branch in setContent which doesn't
    seem to be specifically tested. But again that can be done in another
    ticket.


Please merge after fixing at least 1.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun commented|
|-|-|

Thanks for the review!

> Should there also be a test to show that the temporary file is left behind if there is an IOError during writing or renaming? The docstring for "test_customExtension" describes that, but doesn't actually test it. But it was never tested so we can save that for another ticket I suppose.

[ticket:6780 Nope] :)

> Also there seems to be a windows specific branch in setContent which doesn't seem to be specifically tested. But again that can be done in another ticket.

I think this must be tested on the Windows builders because the tests do call `setContent` and make certain assertions that desirable results were achieved.  I *think* the tests would just fail on Windows without this condition.  It could be a case of extra, unnecessary code though.  Someone would need to delete the conditional and run the tests on Windows to determine if this is the case, I think.  I don't have easy access to Windows so I probably won't do that.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun set status to `closed`|
|-|-|

(In [40404]) Merge filepath-tests-6788

Author: exarkun
Reviewer: rwall
Fixes: [#6788](https://github.com/twisted/twisted/issues/6788)

Improve the factoring of the unit tests for `twisted.python.filepath`.
